jetty-client/src/main/java/org/eclipse/jetty/client/api/Request.java

```diff
@@ -615,4 +634,136 @@ public default void onFailure(Request request, Throwable failure)
         {
         }
     }
+
+    /**
+     * <p>A reactive model to produce request content, similar to {@link java.util.concurrent.Flow.Publisher}.</p>
+     * <p>Implementations receive the content consumer via {@link #subscribe(Consumer, boolean)},
+     * and return a {@link Subscription} as the link between producer and consumer.</p>
+     * <p>Content producers must notify content to the consumer only if there is demand.</p>
+     * <p>Content consumers can generate demand for content by invoking {@link Subscription#demand()}.</p>
+     */
+    public interface Content
```

Comment:
Why don't we use the `Flow.Publisher` API as a base for this rather than create our own near identical version of it?

Reply:
For these reasons: 
* the Flow APIs have to obey the reactivestreams specification and we don't strictly want to do that (e.g. `demand()` does not take a `long`)
* we need some content-specific behavior that cannot be encoded in the simpler interfaces provided by reactivestreams (e.g. `100-Continue`)
* we need to have clear paths for failures that are completely absent from reactivestreams, in both directions (the producer fails, or the consumer fails).
* `Flow.Subscriber.onNext(T)` is not enough, as we would need to allocate all the times an object that wraps `(buffer, last, callback)`